src/support/DatePeriods.js

```diff
@@ -498,6 +498,20 @@ class DatePeriods {
       this.dhis2MonthPeriod(year, month)
     );
   }
+
+  static previousQuarterByPeriodNumber(period, previousPeriods){
+    var previousQuarter = "";
+    var arrayPreviousQuarter = [];
+    for(var i=0; i < previousPeriods; i++){
+      if(i === 0){
+        previousQuarter = this.previousQuarter(period);
```

Comment:
It looks "dirty" in the DatePeriods and we have already a way to ask the "next" and "previous" period
https://github.com/BLSQ/blsq-report-components/blob/39ad3b5b33e60d30a1e943240e9111b0490878b5/src/support/DatePeriods.js#L227
https://github.com/BLSQ/blsq-report-components/blob/39ad3b5b33e60d30a1e943240e9111b0490878b5/src/support/DatePeriods.js#L243
I would reuse that instead calculating quarters and calling previousQuarter

I would not include the current period in the return array of periods
```
static previousPeriods(period, number_of_periods)
static nextPeriods(period, number_of_periods)
```

and add a test for all periods

```
previousPeriods("2018Q3", 3) == ["2017Q4", "2018Q1", "2018Q2"]
previousPeriods("201802", 3) == ["201711", "201712", "201801"]
previousPeriods("2018", 3) == ["2015","2016","2017"]
```
idem for `nextPeriods`